plugins/inx/server_utxo.go

```diff
@@ -216,10 +233,92 @@ func (s *INXServer) ListenToLedgerUpdates(req *inx.LedgerUpdateRequest, srv inx.
 	return ctx.Err()
 }
 
+func (s *INXServer) ListenToTreasuryUpdates(req *inx.LedgerRequest, srv inx.INX_ListenToTreasuryUpdatesServer) error {
+	var sentTreasuryUpdate bool
+	sendPreviousMilestoneDiffs := func(startIndex milestone.Index) error {
+		deps.UTXOManager.ReadLockLedger()
+		defer deps.UTXOManager.ReadUnlockLedger()
+
+		ledgerIndex, err := deps.UTXOManager.ReadLedgerIndex()
+
+		if startIndex > 0 {
+			// Stream all available milestone diffs first
+			pruningIndex := deps.Storage.SnapshotInfo().PruningIndex
+			if startIndex <= pruningIndex {
+				return status.Errorf(codes.InvalidArgument, "given startMilestoneIndex %d is older than the current pruningIndex %d", startIndex, pruningIndex)
+			}
+
+			if err != nil {
+				return status.Error(codes.Unavailable, "error accessing the UTXO ledger")
+			}
+			for currentIndex := startIndex; currentIndex <= ledgerIndex; currentIndex++ {
+				msDiff, err := deps.UTXOManager.MilestoneDiff(currentIndex)
```

Comment:
```suggestion
				msDiff, err := deps.UTXOManager.MilestoneDiffWithoutLocking(currentIndex)
```

Reply:
Also look at the rest of the server code, there is at least another part where we did that mistake. (I can't mark the position because that line is not edited :trollface: )